server/src/main.rs

```diff
@@ -0,0 +1,254 @@
+use std::{thread, time};
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+
+use common::{ChatMessage, Client, Room};
+use hyper::{
+    Body,
+    body::to_bytes,
+    Request, Server, service::{make_service_fn, service_fn},
+};
+use route_recognizer::Params;
+use uuid::Uuid;
+use warp::{Filter, Rejection};
+
+use router::Router;
+
+use crate::utils::{log_msg, setup_app_dir};
+
+mod handler;
+mod router;
+mod ws;
+mod utils;
+
+const WS_ADDR: &str = "127.0.0.1:8000/ws";
+const SERVER_SIGNATURE: &str = "SERVER"; //TODO: zarezerwowanie tego nicka
```

Comment:
Gdybyśmy chcieli potencjalnie rozwinąć ten projekt i klienci mogliby się podłączać do wielu serwerów, wtedy opcja podawania nazw serwera byłaby przydatna (i pewnie do podanej nazwy chcemy dodać jakis odpowiedni prefix, żeby wiedzieć, że to serwer). Ale to tylko hipotetyczna sytuacja.